File: src/com/betterzhang/learnkotlin/java/Student.java
package com.betterzhang.learnkotlin.java;

/**
 * Created by IntelliJ IDEA.
 * Author : Andrew Zhang
 * Email  : betterzhang.dev@gmail.com
 * Time   : 2017/06/28 上午 10:02
 * Desc   : description
 */
public final class Student extends People {

    private String school = null;

    public Student(String name, String school) {
        super(name);
        this.school = school;
    }

    @Override
    public void sayHello() {
        super.sayHello();
        System.out.println("Hello Student");
    }

}

File: src/com/betterzhang/learnkotlin/kotlin/Student.kt
package com.betterzhang.learnkotlin.kotlin

/**
 * Created by IntelliJ IDEA.
 * Author : Andrew Zhang
 * Email  : betterzhang.dev@gmail.com
 * Time   : 2017/06/28 上午 10:17
 * Desc   : description
 */
class Student(private var school: String? = null, name: String) : People(name) {

    override fun sayHello() {
        super.sayHello()
        println("Hello Student")
    }

}